packages/components/src/components/post-collapsible-trigger/post-collapsible-trigger.tsx

```diff
@@ -0,0 +1,55 @@
+import { Component, Element, Prop, Watch } from '@stencil/core';
+import { version } from 'typescript';
+import { checkNonEmpty, checkType } from '@/utils';
+
+@Component({
+  tag: 'post-collapsible-trigger',
+})
+export class PostCollapsibleTrigger {
+  private trigger: HTMLElement | undefined;
+
+  @Element() host: HTMLPostCollapsibleTriggerElement;
+
+  /**
+   * Link the trigger to a collapsible with this id
+   */
+  @Prop() for: string;
+
+  @Watch('for')
+  setAriaAttributes() {
+    checkNonEmpty(this.for, 'The post-collapsible-trigger "for" prop is required.');
+    checkType(this.for, 'string', 'The post-collapsible-trigger "for" prop should be a id.');
+
+    // Add collapsible id to aria-controls
+    if (this.trigger) {
+      this.trigger.setAttribute('aria-controls', this.for);
+
+      const isOpen = !this.collapsible?.collapsed;
+      if (isOpen !== undefined) this.trigger.setAttribute('aria-expanded', `${isOpen}`);
+    }
+  }
+
+  componentWillLoad() {
```

Comment:
In case the button renders slower than the trigger.

```suggestion
  componentDidLoad() {
```